bot/cogs/filtering.py

```diff
@@ -126,6 +135,43 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
+    async def bad_words_in_name(self, msg: Message) -> None:
+        """Check bad words from user display name. When there is more than 3 days after last alert, send new alert."""
+        if not self.name_lock:
+            self.name_lock = asyncio.Lock()
```

Comment:
Isn't this itself a race condition (though not much of an impact)? :) Why aren't you creating the lock immediately, if there's a reason it should probably be documented...

Reply:
I will add comment tomorrow, but lock is not thread safe.

Reply:
The lock doesn't have to be created in an asynchronous context. It can and should be assigned in the cog's `__init__` method. Then the filter can simply rely on it already existing.

Reply:
I had a chat with @SebastiaanZ about my comment and I should clarify:

> It can and should be assigned in the cog's `__init__` method.

The lock must bind to the correct event loop, and so *we* can create it in the cog's `__init__` because `discord.py` will initialise the event loop before the cog is instantiated. It is therefore safe (and convenient) for us to do so, but that doesn't mean that locks can always be instantiated as soon as possible.

Reply:
Okay, resolved with 0e12ff1.